cloudigrade/api/tasks/maintenance.py

```diff
@@ -626,3 +627,63 @@ def migrate_account_numbers_to_org_ids():
             f"Failed to migrate account_numbers to org_ids - {e}", exc_info=True
         )
         return
+
+
+def get_sqs_message_count_cache_key(key):
+    """Get the cache key for an SQS queue's message count."""
+    return f"sqs_message_count_{key}"
+
+
+@shared_task(name="api.tasks.check_and_cache_sqs_queues_lengths")
+def check_and_cache_sqs_queues_lengths():
+    """
+    Check and cache the lengths of specific SQS queues.
+
+    Specifically, check lengths of the houndigrade inspection results queue, its DLQ,
+    the S3 CloudTrail notification queue, and its DLQ. Because these are each treated
+    somewhat differently in cloudigrade, the construction of names and URLs and their
+    processing logic here is somewhat bespoke.
+    """
+    expiration_ttl = settings.CACHED_SQS_QUEUE_LENGTH_EXPIRATION_TTL
+
+    houndigrade_queue_name = settings.HOUNDIGRADE_RESULTS_QUEUE_NAME
+    houndigrade_dlq_name = aws.get_sqs_queue_dlq_name(houndigrade_queue_name)
+    houndigrade_queue_url = aws.get_sqs_queue_url(houndigrade_queue_name)
+    houndigrade_dlq_url = aws.get_sqs_queue_url(houndigrade_dlq_name)
+
+    cloudtrail_queue_url = settings.AWS_CLOUDTRAIL_EVENT_URL
+    cloudtrail_queue_name = cloudtrail_queue_url.split("/")[-1]
```

Comment:
I wonder if we shouldn't have this as a settings.AWS_CLOUDTRAIL_QUEUE_NAME and use it here and in the construction of the AWS_CLOUDTRAIL_EVENT_URL setting.

Reply:
Yeah, I thought about that too, and I was on the fence about whether or not I should also refactor them to be more consistent. Now that I'm pretty confident that the rest of the new code works as expected, I agree this should be simplified, and I'll update related code to use matching config styles. Thanks for nudging me in the right direction!

Reply:
we can revisit this later with a separate PR.